### PR TITLE
Tools: convert autotest build.X and fly.X to use waf build system

### PR DESCRIPTION
**Pull request by peterbarker**

### CODE REVIEW

File: Tools/autotest/arducopter.py [hunk 1]
```diff
@@ -917,10 +917,10 @@ def fly_ArduCopter(viewerip=None, map=False, valgrind=False):
     global homeloc
 
     if TARGET != 'sitl':
-        util.build_SIL('ArduCopter', target=TARGET)
+        util.build_SIL('bin/arducopter-quad', target=TARGET)
```
Comment (OXINARF) [inline Tools/autotest/arducopter.py]: This should be removed, _target_ stops being an option in _build_SIL_

File: Tools/autotest/pysim/util.py [hunk 2]
```diff
@@ -105,35 +120,38 @@ def pexpect_drain(p):
...
+    return (" ".join([ '"%s"' % x for x in cmd ]))
+
+def start_SIL(binary, valgrind=False, gdb=False, wipe=False, synthetic_clock=True, home=None, model=None, speedup=1, defaults_file=None):
     '''launch a SIL instance'''
     import pexpect
-    cmd=""
+    cmd=[]
     if valgrind and os.path.exists('/usr/bin/valgrind'):
-        cmd += 'valgrind -q --log-file=%s-valgrind.log ' % atype
+        cmd.extend(['valgrind', '-q', '--log-file', '%s-valgrind.log' % atype])
```
Comment (OXINARF) [inline Tools/autotest/pysim/util.py]: You have _atype_ here but you don't receive it anymore.

File: Tools/autotest/pysim/util.py [hunk 3]
```diff
@@ -105,35 +120,38 @@ def pexpect_drain(p):
...
-        cmd += ' --model=%s' % model
+        cmd.extend(['--model', model])
     if speedup != 1:
-        cmd += ' --speedup=%f' % speedup
+        cmd.extend(['--speedup', str(speedup)])
     if defaults_file is not None:
-        cmd += ' --defaults=%s' % defaults_file
-    print("Running: %s" % cmd)
-    ret = pexpect.spawn(cmd, logfile=sys.stdout, timeout=5)
+        cmd.extend(['--defaults', defaults_file])
```
Comment (OXINARF) [inline Tools/autotest/pysim/util.py]: Just to make sure: in the previous options the _=_ isn't needed?

File: Tools/autotest/pysim/util.py (hunk 2 above)
Comment (peterbarker) [inline Tools/autotest/pysim/util.py]: On Thu, 26 May 2016, Francisco Ferreira wrote:

> 
> In Tools/autotest/pysim/util.py:
> 
> >      if valgrind and os.path.exists('/usr/bin/valgrind'):
> > -        cmd += 'valgrind -q --log-file=%s-valgrind.log ' % atype
> > +        cmd.extend(['valgrind', '-q', '--log-file', '%s-valgrind.log' % atype])
> 
> You have atype here but you don't receive it anymore.

Darn.  Will I make "binary" shell-safe and use that instead?  I wonder if 
anything relies on that filename...

Comment (OXINARF) [inline Tools/autotest/pysim/util.py]: _binary_ in that function is the path - but you can get the basename. It's only the logfile for valgrind, I doubt that anything relies on it.

Comment (OXINARF) [inline Tools/autotest/pysim/util.py]: Actually it does, an example: https://github.com/ArduPilot/ardupilot/blob/master/Tools/autotest/quadplane.py#L123

Comment (peterbarker) [inline Tools/autotest/pysim/util.py]: On Thu, 26 May 2016, Francisco Ferreira wrote:

> In Tools/autotest/pysim/util.py:
> 
> >      if valgrind and os.path.exists('/usr/bin/valgrind'):
> > -        cmd += 'valgrind -q --log-file=%s-valgrind.log ' % atype
> > +        cmd.extend(['valgrind', '-q', '--log-file', '%s-valgrind.log' % atype])
> 
> You have atype here but you don't receive it anymore.

I've pushed up some changes to fix this.  A SIL object now gives us the 
path.

File: Tools/autotest/arducopter.py (hunk 1 above)
Comment (peterbarker) [inline Tools/autotest/arducopter.py]: On Thu, 26 May 2016, Francisco Ferreira wrote:

> In Tools/autotest/arducopter.py:
> 
> > @@ -917,10 +917,10 @@ def fly_ArduCopter(viewerip=None, map=False, valgrind=False):
> >      global homeloc
> > 
> >      if TARGET != 'sitl':
> > -        util.build_SIL('ArduCopter', target=TARGET)
> > +        util.build_SIL('bin/arducopter-quad', target=TARGET)
> 
> This should be removed, target stops being an option in build_SIL

Done.

File: Tools/autotest/pysim/util.py (hunk 3 above)
Comment (peterbarker) [inline Tools/autotest/pysim/util.py]: On Thu, 26 May 2016, Francisco Ferreira wrote:

> In Tools/autotest/pysim/util.py:
> 
> >      if defaults_file is not None:
> > -        cmd += ' --defaults=%s' % defaults_file
> > -    print("Running: %s" % cmd)
> > -    ret = pexpect.spawn(cmd, logfile=sys.stdout, timeout=5)
> > +        cmd.extend(['--defaults', defaults_file])
> 
> Just to make sure: in the previous options the = isn't needed?

Correct.

**Comment by lucasdemarchi** [comment]
Applied, thanks.

**Pull request closed without merging**